aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -53,6 +80,18 @@ static inline Tensor unary_op_impl(const Tensor& self, OutImpl& out_impl) {
   return out_impl(result, self);
 }
 
+template <typename OutImpl>
+static inline Tensor unary_op_impl_complex_to_float(const Tensor& self, OutImpl& out_impl) {
+  if (self.is_complex()) {
+    const auto float_type = c10::toValueType(self.scalar_type());
+    Tensor result = at::empty({0}, self.options().dtype(float_type));
+    return out_impl(result, self);
+  }
+
+  Tensor result = at::empty({0}, self.options());
+  return out_impl(result, self);
```

Comment:
why do we need this? don't we only call this function for complex tensors?

Reply:
ahhh I see. you are calling this method from abs for non complex tensors as well. I think it's a little misleading to call this method for non complex inputs because of its name. So, I think we can do two things here:
1. change the name of the function to make it more generic. 
2. call this method for only complex inputs.

Whatever you prefer.

Reply:
maybe just add a word or two like
`unary_op_impl_with_complex_to_float`

Reply:
Good idea.

Reply:
I still think it will not be intuitive to a developer who is not familiar with this PR and looks at this code and it could be confusing. I would caution on the side of either removing "complex_to_float" from the name or adding descriptive comments specifically mentioning this function is also being used for real values.

Reply:
I updated the comment.